### PR TITLE
Fix aggregation query parameters to correctly replace keys in list

### LINKED ISSUE #1025: Aggregation query parameter does not replace keys in the lists
I was working on an aggregation endpoint:

```python
things_recommended = {
    'url': 'things/recommended/',
    'datasource': {
        'source': 'things',
        'aggregation': {
            'pipeline': [
                {"$match": {"$or": [{"family": "$family_id"}, {"is_shared": True}]}},
                {"$addFields": {
                    "recommended_for_me":
                    {"$setIsSubset":[["$user_id"], "$recommended_to"]}
                }},
                {"$sort": SON([("recommended_for_me", -1), ("happened", -1)])}
            ]
        }
    }
}
```

However, I found that `$family_id` and `$user_id` are not getting replaced since the code only covers the case of a pure dictionary without lists.

See: https://github.com/pyeve/eve/blob/22ea4bfebc8b633251cd06837893ff699bd07a00/eve/methods/get.py#L126

I'm new to Eve, so I'm not sure that what I found worth an error report. Though it took me quite a time to debug.

Best regards,
Youngjae

PS, Eve is AWESOME.

### PR DESCRIPTION
**Pull request by yjcrocks**
This commit resolves #1025.

### CODE REVIEW

File: eve/methods/get.py [hunk 1]
```diff
@@ -122,11 +122,18 @@ def _perform_aggregation(resource, pipeline, options):
...
-                parse_aggregation_stage(st_value, key, value)
-            if key == st_value:
-                d[st_key] = value
+        if isinstance(d, dict):
+            for st_key, st_value in d.items():
+                if isinstance(st_value, dict) or isinstance(st_value, list):
+                    parse_aggregation_stage(st_value, key, value)
+                if key == st_value:
+                    d[st_key] = value
+        if isinstance(d, list):
```
Comment (Amedeo91) [inline eve/methods/get.py]: elif

File: eve/methods/get.py [hunk 2]
```diff
@@ -122,11 +122,18 @@ def _perform_aggregation(resource, pipeline, options):
...
 
     def parse_aggregation_stage(d, key, value):
-        for st_key, st_value in d.items():
-            if isinstance(st_value, dict):
-                parse_aggregation_stage(st_value, key, value)
-            if key == st_value:
-                d[st_key] = value
+        if isinstance(d, dict):
+            for st_key, st_value in d.items():
+                if isinstance(st_value, dict) or isinstance(st_value, list):
```
Comment (Amedeo91) [inline eve/methods/get.py]: Can you create a method in order to avoid duplication?

File: eve/methods/get.py [hunk 3]
```diff
@@ -122,11 +122,18 @@ def _perform_aggregation(resource, pipeline, options):
...
+                    parse_aggregation_stage(st_value, key, value)
+                if key == st_value:
+                    d[st_key] = value
+        if isinstance(d, list):
+            for st_idx, st_value in enumerate(d):
+                if isinstance(st_value, dict) or isinstance(st_value, list):
+                    parse_aggregation_stage(st_value, key, value)
+                if key == st_value:
+                    d[st_idx] = value
```
Comment (Amedeo91) [inline eve/methods/get.py]: Let us see what @nicolaiarocci says, but maybe it is worth to report it in the documentation related to the aggregation endpoint

**Comment by Amedeo91** [comment]
Hi,

Can you please squash the commit into one? Btw, the continuos integration is failing! :)

Regards,
Amedeo

**Comment by nicolaiarocci** [comment]
Hello, this is very nice. I am willing to merge. Would you also provide test coverage?

**Comment by idserge7** [comment]
Do you need some help with this issue? I've already experimented with aggregation parameters and also with adding pre_aggregation and on_fetched_aggregation hooks. Currently also without test coverage but I can write them.

**Comment by Amedeo91** [comment]
@idserge7: without test coverage I do not think @nicolaiarocci is willing to merge! But please, do your PR! :)

**Comment by idserge7** [comment]
I'll cover it with tests first :)

**Comment by nicolaiarocci** [comment]
Is anyone on test coverage for this?

**Comment by nicolaiarocci** [comment]
Nevermind we have a similar PR with proper testing. Will go for that one instead: #1058

**Pull request closed without merging**